### ISSUE TITLE
Stringsdict synthesized accessors generation avoid non-plural arguments

### ISSUE THREAD

**Issue by Bogdan-Belogurov** [tuist/tuist#6379]
### What problem or need do you have?

Hello, thank you for your work! ❤️ 

Currently, we use R.swift and are trying to migrate to the Tuist generation solution but we have faced a problem.

### Context

For example, we have the plural string:
```
<key>key</key>
    <dict>
        <key>NSStringLocalizedFormatKey</key>
        <string>%#@value@</string>
        <key>value</key>
        <dict>
            <key>NSStringFormatSpecTypeKey</key>
            <string>NSStringPluralRuleType</string>
            <key>NSStringFormatValueTypeKey</key>
            <string>d</string>
            <key>one</key>
            <string>Love %1$d item in the %2$s store.</string>
            <key>other</key>
            <string>Love %1$d items in the %2$s store.</string>
        </dict>
    </dict>
```

And Tuist generates an argument only for the `NSStringFormatValueTypeKey`

```
  /// Plural format key: "%#@value@"
  public static func key(_ p1: Int) -> String {
    return ReachViewStrings.tr("Localizable", "key", p1)
  }
```

But expected
```
  /// Plural format key: "%#@value@"
  public static func key(_ p1: Int, p2: Any) -> String {
    return ReachViewStrings.tr("Localizable", "key", p1, p2)
  }
```
I understand that we should use `NSStringLocalizedFormatKey` for this purposes 
like this:
```
<key>key</key>
    <dict>
        <key>NSStringLocalizedFormatKey</key>
        <string>Love %#@value@ in the %2$s store.</string>
        <key>value</key>
        <dict>
            <key>NSStringFormatSpecTypeKey</key>
            <string>NSStringPluralRuleType</string>
            <key>NSStringFormatValueTypeKey</key>
            <string>d</string>
            <key>one</key>
            <string>%1$d item</string>
            <key>other</key>
            <string>%1$d items</string>
        </dict>
    </dict>
```
But for our dev flow, it's hard to do that. Maybe there is a way around this problem? 🙏🏿 

### macOS version

14.3.1

### Tuist version

4.12.1

### Xcode version

15.3

**Comment by pepicrft** [comment]
HI @Bogdan-Belogurov 👋🏼 
It's fantastic to hear that you are adopting Tuist, and we'd be super happy to support you in the process. Tuist uses SwiftGen under the hood with the templates in [this directory](https://github.com/tuist/tuist/tree/main/Sources/TuistGenerator/Templates). Based on your description, it seems that the string template does not cover your use case. Can you spot what might be wrong with that template? We'd be happy to update it with the fix, or even better, you can go ahead and open a PR yourself. Alternatively, you can [provide your own templates](https://docs.tuist.io/guide/project/synthesized-files.html#resource-accessors) and align them with the conventions that you were previously following with R.swift.

**Comment by Bogdan-Belogurov** [comment]
@pepicrft 
Thanks for the reply! I'll try to improve the template, thanks for the clue ❤️

**Comment by fortmarek** [comment]
> What can you recommend? I can try to fix it and make a PR, but there are a lot of old and stale PRs and I'm not sure it will be quick. Or I can try to make a template like proposed in the issue 🤔

Yes, creating a PR to `swiftgen` would be ideal. As things would have it, I actually have rights to merge PRs in the `SwiftGen` repository. I haven't contributed in quite a while, though, so I would need to contact one of the contributors to get an additional approval before merging anything.

In the worst case, we can depend on a `tuist` fork.
